dali/pipeline/operators/util/external_source.cc

```diff
@@ -29,10 +29,13 @@ void ExternalSource<CPUBackend>::RunImpl(SampleWorkspace *ws, const int idx) {
     output.Copy(data, stream);
   }
 
-  std::unique_lock<std::mutex> l(samples_processed_m_);
+  std::lock_guard<std::mutex> l(samples_processed_m_);
   if (++samples_processed_ >= batch_size_) {
```

Comment:
Since it's inside a lock, samples_processed_ can be a regular variable, not atomic one.